### PR TITLE
fix: skip generating error events for unauthorized or forbidden errors

### PR DESCRIPTION
**Pull request by duyhungtnn**
part of https://github.com/bucketeer-io/bucketeer/issues/1284

Screenshot 2024-11-22 at 15 49 33

### CODE REVIEW

**Comment by duyhungtnn** [comment]
@cre8ivejp please help me to take a look

File: src/objects/metricsEvent.ts [hunk 1]
```diff
@@ -197,9 +197,11 @@ export const toErrorMetricsEvent = (e: any, tag: string, apiId: NodeApiIds): Eve
       case statusCode == 400:
         return createBadRequestErrorMetricsEvent(tag, apiId);
       case statusCode == 401:
-        return createUnauthorizedErrorMetricsEvent(tag, apiId);
+        console.log('An unauthorized error occurred. Please check your API Key.');
```
Comment (cre8ivejp) [inline src/objects/metricsEvent.ts:200]: How about using the error level?

```suggestion
        console.error('An unauthorized error occurred. Please check your API Key.');
```

File: src/objects/metricsEvent.ts [hunk 2]
```diff
@@ -197,9 +197,11 @@ export const toErrorMetricsEvent = (e: any, tag: string, apiId: NodeApiIds): Eve
...
+        return null;
       case statusCode == 403:
-        return createForbiddenErrorMetricsEvent(tag, apiId);
+        console.log('An forbidden error occurred. Please check your API Key.');
```
Comment (cre8ivejp) [inline src/objects/metricsEvent.ts:203]: Same here.

```suggestion
        console.error('An forbidden error occurred. Please check your API Key.');
```

**Comment by duyhungtnn** [comment]
@cre8ivejp I updated to use console.error()

**Review by cre8ivejp — APPROVED** [review]
Thank you!

**Pull request merged**